packages/salesforcedx-vscode-core/src/commands/forceRenameLightningComponent.ts

```diff
@@ -132,9 +132,19 @@ async function renameComponent(sourceFsPath: string, newName: string) {
   notificationService.showWarningMessage(nls.localize(RENAME_WARNING));
 }
 
+export function getParentDirectoryOfTestFolder(sourceFsPath: string, pathSeparator: string): string {
+  const directories = sourceFsPath.split(pathSeparator);
+  directories.splice(directories.length - 1);
+  return directories.join(pathSeparator);
+}
+
 async function getComponentPath(sourceFsPath: string): Promise<string> {
   const stats = await fs.promises.stat(sourceFsPath);
-  return stats.isFile() ? path.dirname(sourceFsPath) : sourceFsPath;
+  let dirname = stats.isFile() ? path.dirname(sourceFsPath) : sourceFsPath;
+  if (dirname.endsWith(TEST_FOLDER)) {
+    dirname = getParentDirectoryOfTestFolder(dirname, path.sep);
```

Comment:
This is where i got tripped up with Windows vs OSX in unit testing.  How you suggested was my original implementation.  I'd appreciate your thoughts on it.

If I use path.sep inside the function itself instead of passing it in, it works great at runtime.  The path that is generated by the OS can be parsed using path.sep.

But during testing, i wanted to unit test that this function works on both platforms. if I want to pass in a string like this `force-app/main/default/lwc/helloworld/__tests__` and verify that it get parsed and the `__tests__` directory removed, the test only passes on my OSX operating system and fails on our Windows CI.  Because on Windows the function under unit test is now looking for `\` and the unit test fails.

I thought about generating the string in the unit test dynamically like `force-app${path.sep}main${path.sep}default${path.sep}lwc${path.sep}helloworld${path.sep}__tests__`.  And maybe that's better, but it just seemed weird that a unit test would run differently depending on where it was executed.

So anyway, I opted to pass in the path.sep into the function, and this allowed me to unit test the function for both operating systems, and it's easy to read the unit test.

So options are:
1. Leave it as is ( pass the separator into the function )
2. Generate the path dynamically in the unit test
3. Use an if/else in the unit test.  Like `if ( path.sep === '/' ) {} else {}` inside the it statement of the unit test based on environment.

Reply:
I looked trough our codebase, but I don't see many places where we are creating platform-specific tests.  I just ran through the exercise of, "what tests would I write if I had written getParentDirectoryOfTestFolder()"?

I started off with just the implementation:
```
export function getParentDirectoryOfTestFolder(sourceFsPath: string): string {
  const directories = sourceFsPath.split(path.sep);
  directories.splice(directories.length - 1);
  return directories.join(path.sep);
}
```

...and then thought about writing a test for it.  This is what immediately came to mind:
```
it('validates getParentDirectoryOfTestFolder', () => {
    const folders = ['path', 'to', 'tests']; // mocked(fake) path
    const folderPath = folders.join(path.sep);

    const parentFolder = folders.slice(0, -1);
    const parentFolderPath = parentFolder.join(path.sep);

    const parentDirectory = getParentDirectoryOfTestFolder(folderPath);
    expect(parentDirectory).to.equal(parentFolderPath)
  });
```

With this test, it validates the logic & behavior inside the function, and leaves the implementation as a black box.

OK, so, thinking out loud here... I could then see one saying that my test is just mirroring the implementation of what's in `getParentDirectoryOfTestFolder()`, which I would say, that's OK, we just want to validate that it works. 
 But if that wan't acceptable, the the next thing I came up with was a test like this:
```
it('validates getParentDirectoryOfTestFolder', () => {
    const folderPath = 'path|to|tests'.replace(/\|/g, path.sep);
    const parentFolderPath = 'path|to'.replace(/\|/g, path.sep);

    const parentDirectory = getParentDirectoryOfTestFolder(folderPath);
    expect(parentDirectory).to.equal(parentFolderPath)
  });
```